### ISSUE TITLE
No resubscribe after subscribe -> unsubscribe -> subscribe to a same channel

### ISSUE THREAD

**Issue by nmorel** [centrifugal/centrifuge-js#62]
When unsubscribing to a channel, the property `_noResubscribe` is set to true on `Sub` object.  
The subscription is kept around and when subscribing to the same channel, the subscription is reused but the property `_noResubscribe` is still true.   
It prevents the resubscription mechanism from working.

**Comment by FZambia** [comment]
The same here - code snippet will help to reproduce and fix problem. From what I understand we need to add `this._noResubscribe = false` around [this line](https://github.com/centrifugal/centrifuge-js/blob/master/src/centrifuge.js#L1724) - i.e. on subscribe called. Do I understand right?

**Comment by nmorel** [comment]
```javascript
const channel = 'test';

const centrifuge = new Centrifuge({
  "resubscribe": true,
  ...
});
centrifuge.connect();

let subscription = centrifuge.subscribe(channel);
// Here, if a network problem happens, the resubscribe works

// Now we unsubscribe
subscription.unsubscribe();

// And later, subscribe to the same channel
subscription = centrifuge.subscribe(channel);

// Here, if a network problem happens, the resubscribe no longer works
```

Your commit should fix the issue yes :)

**Comment by FZambia** [comment]
Thanks! Could you try version with fix in your app and confirm that it fixes an issue?

**Comment by FZambia** [comment]
@nmorel I just looked at my commit and realized that I set `_noResubscribe` property in the wrong place:) Can't explain what I was thinking about at that moment. Just pushed a fix into master in 12a47d24a8d2e3d51958d109dfd3a826a5d2fc16

**Comment by nmorel** [comment]
I tested your last fix, it works for the resubscription but I have other issues now :)  
I have recover enabled on my server.   

When I subscribe to a channel I unsubscribed, it is considered the same as a resubscription (after a network failure for example).  
The properties `last` and `recover` are sent in the subscribe message and the context given in subscribe event has `isResubscribe` to true.  

There are probably a few properties that should be reset at the same time you set `_noResubscribe` to false like `_isResubscribe` or the `_lastMessageID`.

```javascript
const channel = 'test';

const centrifuge = new Centrifuge({
  "resubscribe": true,
  ...
});
centrifuge.connect();

let subscription = centrifuge.subscribe(channel);

// Now we unsubscribe
subscription.unsubscribe();
subscription.removeAllListeners()

// And later, subscribe to the same channel
subscription = centrifuge.subscribe(channel, {
  subscribe(ctx) {
    // ctx.isResubscribe is true!
  }
});
// It sends last and recover properties as if it was a resubscription and the context returned in subscribe event indicates it is a resubscribe
```

**Comment by nmorel** [comment]
It works, thanks!

**Comment by nmorel** [comment]
Nvm, the `isResubscribe` flag is still true.  
But the recover and last properties are not sent.

**Comment by FZambia** [comment]
BTW could you describe for what purposes do you use `isResubscribe` flag in your app?

**Comment by nmorel** [comment]
I use it to reload my state, users must not miss messages.

**Comment by FZambia** [comment]
Ah I see, looks like you are using it instead of `recovered` because of the fact that `recovered` returns some false positives in some cases - makes sense.

**Comment by nmorel** [comment]
Actually, I'm using both :)   
```javascript
function onSubscribe(ctx) {
  if (ctx.isResubscribe && !ctx.recovered) {
    // reload state
  }
}
```

**Comment by FZambia** [comment]
@nmorel got it:) This time I will reproduce this case on local computer - looks like this is not that simple I thought:)

**Comment by FZambia** [comment]
@nmorel just made a branch with possible fix - https://github.com/centrifugal/centrifuge-js/tree/fix_is_resubscribe - hope this time this will work as expected. Also backported everything to `c2` branch.

Btw - I also checked that in `c2` branch with Centrifugo v2 short disconnect and resubscribe to channel without any history results in `recovered` flag set to `true`.

**Comment by nmorel** [comment]
Your last fix is working for me, thanks!  

Good to know for v2 :)

**Comment by FZambia** [comment]
Released in 1.5.0
